sw/device/tests/spi_host_smoketest.c

```diff
@@ -151,6 +151,24 @@ status_t test_quad_read(void) {
   return OK_STATUS();
 }
 
+status_t test_4bytes_address(void) {
```

Comment:
Should we read out the SPI flash capabilities and skip this test if 4-byte addressing is not supported?

We're already reading out the SFDP in this test to figure out how to enable quad mode.  In the basic flash parameters table, in the first 32-bit word, bits 17:19 contains the supported address modes:
```
0 - 3b mode only
1 - 3b and 4b modes supported
2 - 4b mode only
3 - reserved
```

Reply:
Sounds good to me.